src/Faker/Extension/Helper.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Faker\Extension;
+
+/**
+ * A class with some methods that may make building extensions easier.
+ */
+final class Helper
```

Comment:
Hmm, why not leave the basic methods in `Faker\Generator`?

Extensions can compose an instance of `Faker\Generator` and use these methods, then.

Reply:
That is very true. 

But I decided not to use `Generator` here for a few reasons: 
1. They do not belong to "Generator" since they are not generating anything.
2. These methods are common around extensions and it will reduce complexity if we could avoid ´GeneratorAwareExtension´
3. It will be easier to test them
4. The `Generator` (will) have many many method as it is.

Reply:
> 1. They do not belong to "Generator" since they are not generating anything.

That's not the case for `numerify()`, `lexify()`, and `bothify()` - they all generate `string`s based on other `string`s.

> 2. These methods are common around extensions and it will reduce complexity if we could avoid `GeneratorAwareExtension`

Do you want to drop the `GeneratorAwareExtension` interface entirely then?

> 3. It will be easier to test them

How will it be easier to test them when all of these methods are `static`?

> 4. The Generator (will) have many many method as it is.

Reply:
1. Hm maybe. 
2. No, it is still good if you want to generate combined things. Ie, the WelcomeMessageExtension combines "$generator->name()" with "$generator->readText()"
3. You dont need to mock the Generator, If you want to test the AddressExtension you only need to test the output with a regex.

Reply:
I been thinking of this. It is 50/50 what to do. 

But I want the `Generator` to be stupid and not be able to do anything without its extensions. That is why I dont want to add this logic in the Generator. I also think that this `Helper` class is almost feature complete. We may want to add one more maximum two more methods in there, but if we do more than that, I feel like we are doing things wrong.